components/dms/MessageActorMeta.tsx

```diff
@@ -0,0 +1,145 @@
+import { Alert } from '@/common/Alert'
+import { ComponentProps, useState } from 'react'
+import client from '@/lib/client'
+import { useQuery } from '@tanstack/react-query'
+import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/24/solid'
+import { ChatBskyModerationGetActorMetadata } from '@atproto/api'
+
+const MockData: ChatBskyModerationGetActorMetadata.OutputSchema = {
+  day: {
+    messagesSent: 25,
+    messagesReceived: 30,
+    convos: 10,
+    convosStarted: 5,
+  },
+  month: {
+    messagesSent: 700,
+    messagesReceived: 650,
+    convos: 300,
+    convosStarted: 150,
+  },
+  all: {
+    messagesSent: 1500,
+    messagesReceived: 1450,
+    convos: 800,
+    convosStarted: 400,
+  },
+}
+
+export const useMessageActorMeta = ({
+  did,
+  enabled,
+}: {
+  did: string
+  enabled: boolean
+}) => {
+  // This query is a bit expensive but the data does change relatively frequently so we wanna cache it for only 10m
+  return useQuery<
+    unknown,
+    unknown,
+    Record<string, ChatBskyModerationGetActorMetadata.Metadata>
+  >({
+    cacheTime: 10 * 60 * 1000,
+    staleTime: 10 * 60 * 1000,
+    retry: 0,
+    enabled,
+    queryKey: ['messageActorMeta', { did }],
+    queryFn: async () => {
+      // TODO: Use this instead of the mock data
+      // const { data } = await client.api.chat.bsky.moderation.getActorMetadata(
+      // { actor: did },
+      // { headers: client.proxyHeaders() },
```

Comment:
We'll want the proxy headers to reference the backing chat service.

Reply:
Oops, forget I said that!  The request will go through ozone, which will perform service auth with the chat service.  So this looks correct 👍